profiler/src/main/java/com/splunk/opentelemetry/profiler/JfrActivator.java

```diff
@@ -67,34 +69,69 @@ public class JfrActivator implements AgentListener {
   @Override
   public void afterAgent(AutoConfiguredOpenTelemetrySdk autoConfiguredOpenTelemetrySdk) {
     ConfigProperties config = autoConfiguredOpenTelemetrySdk.getConfig();
+    if (notClearForTakeoff(config)) {
+      return;
+    }
+
+    configurationLogger.log(config);
+    logger.info("JFR profiler is active.");
+    executor.submit(
+        logUncaught(
+            () -> activateJfrAndRunForever(config, autoConfiguredOpenTelemetrySdk.getResource())));
+  }
+
+  private boolean notClearForTakeoff(ConfigProperties config) {
     if (!config.getBoolean(CONFIG_KEY_ENABLE_PROFILER, false)) {
       logger.fine("xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx");
       logger.fine("xxxxxxxxx  JFR PROFILER DISABLED!  xxxxxxxxx");
       logger.fine("xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx");
-      return;
+      return true;
     }
     if (!JFR.instance.isAvailable()) {
       logger.warning(
           "JDK Flight Recorder (JFR) is not available in this JVM. Profiling is disabled.");
-      return;
+      return true;
     }
-    configurationLogger.log(config);
-    logger.info("JFR profiler is active.");
-    executor.submit(
-        logUncaught(
-            () -> activateJfrAndRunForever(config, autoConfiguredOpenTelemetrySdk.getResource())));
+
+    Path outputDir = Paths.get(config.getString(CONFIG_KEY_PROFILER_DIRECTORY));
+    if (!Files.exists(outputDir)) {
+      // Try creating the directory for the user...
+      try {
+        Files.createDirectories(outputDir);
+      } catch (IOException e) {
+        return outdirWarn(outputDir, "does not exist and could not be created");
+      }
+    }
+    if (!Files.isDirectory(outputDir)) {
+      return outdirWarn(outputDir, "exists but is not a directory");
+    }
+
+    if (!Files.isWritable(outputDir)) {
+      return outdirWarn(outputDir, "exists but is not writable.");
+    }
+    return false;
+  }
+
+  private boolean outdirWarn(Path dir, String suffix) {
+    logger.warning(
+        "PROFILER WILL NOT BE STARTED: The configured output directory "
+            + dir
+            + " "
+            + suffix
+            + ".");
```

Comment:
Should we use the string template here?
```suggestion
    logger.log(
        WARNING,
        "PROFILER WILL NOT BE STARTED: The configured output directory {0} {1}.",
        new Object[] {dir, suffix});
```